tests/js/server/paths/filesystem.js

```diff
@@ -0,0 +1,350 @@
+    let content = fs.adler32(fn);
+    assertTrue(content > 0);
+  }
+  function tryWriteAllowed(fn, text) {
+    let rc = fs.write(fn, text);
+    assertTrue(rc, 'Expected ' + fn + ' to be writeable');
+  }
+
+  function tryAppendAllowed(fn, text) {
+    let rc = fs.append(fn, text);
+    assertTrue(rc, 'Expected ' + fn + ' to be appendeable');
+  }
+
+  function tryChmodAllowed(fn) {
+    let rc = fs.chmod(fn, '0755');
+    assertTrue(rc, 'Expected ' + fn + ' to be chmodable');
+  }
+
+  function tryExistsAllowed(fn, exists) {
+    let rc = fs.exists(fn);
+    assertEqual(rc, exists, 'Expected ' + fn + ' to be stat-eable');
+    if (exists) {
+      rc = fs.mtime(fn);
+      assertTrue(rc > 0);
+    }
+  }
+
+  function tryFileSizeAllowed(fn) {
+    let rc = fs.size(fn);
+    assertTrue(rc, 'Expected ' + fn + ' to be stat-eable');
+  }
+
+  function tryIsDirectoryAllowed(fn, isDirectory) {
+    let rc = fs.isDirectory(fn);
+    assertEqual(rc, isDirectory, 'Expected ' + fn + ' to be a ' + isDirectory ? "directory.":"file.");
+  }
+
+  function tryCreateDirectoryAllowed(fn) {
+    let rc = fs.makeDirectory(fn);
+    assertUndefined(rc, 'Expected ' + fn + ' to become a directory.');
+    tryIsDirectoryAllowed(fn, true);
+  }
+
+  function tryRemoveDirectoryAllowed(fn) {
+    let rc = fs.removeDirectory(fn);
+    assertUndefined(rc, 'Expected ' + fn + ' to become a directory.');
+    tryIsDirectoryAllowed(fn, false);
+  }
+
+  function tryCreateDirectoryRecursiveAllowed(fn) {
+    let rc = fs.makeDirectoryRecursive(fn);
+    assertUndefined(rc, 'Expected ' + fn + ' to become a directory.');
+    tryIsDirectoryAllowed(fn, true);
+  }
+
+  function tryRemoveDirectoryRecursiveAllowed(fn) {
+    let rc = fs.removeDirectoryRecursive(fn);
+    assertUndefined(rc, 'Expected ' + fn + ' to be removed.');
+    tryIsDirectoryAllowed(fn, false);
+  }
+
+  function tryIsFileAllowed(fn, isFile) {
+    let rc = fs.isFile(fn);
+    assertEqual(rc, isFile, 'Expected ' + fn + ' to be a ' + isFile ? "file.":"directory.");
+  }
+
+  function tryListFileAllowed(dn, expectCount) {
+    let rc = fs.list(dn);
+    assertEqual(rc.length, expectCount, 'Expected ' + dn + ' to contain ' + expectCount + " files.");
+  }
+
+  function tryListTreeAllowed(dn, expectCount) {
+    let rc = fs.listTree(dn);
+    assertEqual(rc.length, expectCount, 'Expected ' + dn + ' to contain ' + expectCount + " files.");
+  }
+
+  function tryGetTempFileAllowed(dn) {
+    let tfn = fs.getTempFile();
+    fs.write(tfn, "hello");
+    assertTrue(fs.isFile(tfn));
+    fs.remove(tfn);
+  }
+
+  function tryMakeAbsoluteAllowed(fn) {
+    fs.makeAbsolute(fn);
+  }
+  function tryCopyFileAllowed(sn, tn) {
+    try {
+      fs.copyFile(sn, tn);
+    } catch (err) {
+      assertTrue(false, "failed to copy " + sn + " to " + tn + " - " + err);
+    }
+    tryExistsAllowed(sn, true);
+    tryExistsAllowed(tn, true);
+  }
+  function tryCopyRecursiveFileAllowed(sn, tn) {
+    try {
```

Comment:
same (also true for the next few occurrences)

Reply:
the reason I did this originally is to easily have the combination of source file, destination file and error message in one string. Else you may need to search and (errnously) calculate the filename you believe that was copied.

NB: this function isn't used below.